feat: can override scope

NTARelix opened the pull request:
Here at Calabrio we've had to make a few modifications to this library to get our ForgeRock implementation working. This pull request contains one of the changes we've made. We're currently using these changes successfully. These particular changes allow us to specify additional scope beyond the default: `...&scope=openid email calabrio&...` vs the default `...&scope=openid&...`.

jakefeasel commented:
Thanks for your PR! Can you please describe the use-case you're trying to solve a bit further. Since the response type is id_token, the only difference additional scopes will introduce is additional claims in the id_token. These claims are intended to be compared with the claims initially gathered when the app session starts, to detect some intrinsic difference in the session (which might justify taking an action in response to those differences). Can you explain how these additional scopes will be used for this purpose, for you?

hadimc commented:
@jakefeasel  Thanks for your note. As you suggested, supporting custom scopes (hence passing custom OIDC claims in the ID_Token) would allow an application to do a more detailed comparison of such information. For example, in our world of multi-tenant SaaS solution, a same user can earn different session variables (such as what application tenant they have signed into) goring through a same FR authentication flow. If FR session (including tenant info) changes, while user is using some app services, the application needs to receive this information (via ID_Token) and acts accordingly (deny access, redirect for new authentication, redirect to target tenant Url, etc)... In these cases, the openid scope does not provide enough information. Hope that helps.

jakefeasel reviewed (changes requested):
The code change looks good, but please make the changes to sessionCheck.js rather than sessionCheckGlobal.js. sessionCheckGlobal.js is generated from `npm run build`, and so should not be directly edited. Also, could you please include an update to the README.md to include this new parameter? Thanks

jakefeasel commented:
The new commit for this issue looks great, thanks. I think you need to rebase on master, though - this PR has a duplicate commit from #12 that has already been merged. Please remove that one and I will merge this.

NTARelix commented:
My bad. I thought I had updated my fork with `upstream/master`, but I rebased instead of merge w/ fast-forward so it came up as a separate commit. It appears to be fixed now.

jakefeasel commented:
Thanks for your contribution!
